core/src/main/scala/kafka/controller/ControllerContext.scala

```diff
@@ -259,6 +260,11 @@ class ControllerContext {
     livePreferredControllerIds = preferredControllerIds
   }
 
+  def setCorruptedBrokers(brokers: Map[Int, Boolean]): Unit = {
```

Comment:
This probably has to be synchronized.

Reply:
Agree with Greg's comment above. This is being read only from the event loop, and written either during controller failover (also part of the event loop), and when ZK node is changed (this is also handled through the event loop).

Thus, I do not feel that there is a chance of multiple threads accessing this.